source/common/http/http1/conn_pool.h

```diff
@@ -80,6 +80,21 @@ class ConnPoolImpl : Logger::Loggable<Logger::Id::pool>, public ConnectionPool::
 
     void onConnectTimeout();
 
+    void onIdleTimeout() { parent_.onIdleTimeout(*this); }
+
+    void resetIdleTimer() {
+      if (idle_timer_) {
+        idle_timer_->disableTimer();
+        idle_timer_.reset();
```

Comment:
I think you probably shouldn't delete the timer here; just disable it, so that it can be re-used on the next idle period. And rename the function to disableIdleTimer.

If you delete it here, that means you only get to check for idleness after the first request, but not after subsequent requests (because you never re-create it)